lib/msal-browser/docs/performance.md

```diff
@@ -50,3 +50,80 @@ const msalInstance = new msal.PublicClientApplication({
     }
 });
 ```
+
+## Measuring performance
+
+Applications that want to measure the performance of authentication flows in MSAL.js can do so manually, or consume the performance measures taken by the library itself.
+
+### addPerformanceCallback
+
+Since version `@azure/msal-browser@2.23.0`, applications can register a callback to receive performance measurements taken by the library. These measurement will include end-to-end measurements for top-level APIs, as well as measurements for important internal APIs.
+
+**Note for MSFT first-party applications**: We will be publishing an internal build of `@azure/msal-browser` that is already instrumented to capture this telemetry. Contact us for more details.
+
+#### Example
+
+```typescript
+const msalInstance = new PublicClientApplication(config);
+
+msalInstance.addPerformanceCallback((events: PerformanceEvent[]) => {
+    events.forEach(event => {
+        console.log(event);
+    });
+});
+```
+
+Example event:
+
+```typescript
+const event: PerformanceEvent = {
```

Comment:
could be worth including appName and appVersion from the config as part of this?

Reply:
Sure